sql/driver/conn.go

```diff
@@ -221,15 +228,154 @@ func (c *conn) Delete(p *parser.Delete, args []driver.Value) (*rows, error) {
 }
 
 func (c *conn) Insert(p *parser.Insert, args []driver.Value) (*rows, error) {
-	return nil, fmt.Errorf("TODO(pmattis): unimplemented: %T %s", p, p)
+	desc, err := c.getTableDesc(p.Table)
+	if err != nil {
+		return nil, err
+	}
+
+	// Determine which columns we're inserting into.
+	cols, err := c.processColumns(desc, p.Columns)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct a map from column ID to the index the value appears at within a
+	// row.
+	colMap := map[uint32]int{}
+	for i, c := range cols {
+		colMap[c.ID] = i
+	}
+
+	// Verify we have at least the columns that are part of the primary key.
```

Comment:
We could be inserting a couple of columns to an existing row. I suppose you take that as one of the many TODO